providers/sendgrid/src/lib/sendgrid.provider.spec.ts

```diff
@@ -4,6 +4,7 @@ import { SendgridEmailProvider } from './sendgrid.provider';
 const mockConfig = {
   apiKey: 'SG.1234',
   from: 'test@tet.com',
+  senderName: 'test',
```

Comment:
Should we also update the `mockNovuMessage` to reflect the change?

Reply:
We can add it the senderName to `IEmailOptions` as well and propagate to all integrations but then we assume it's used in every provider, now this change is limited to Sendgrid only. What do you think?

Reply:
@p-fernandez

Reply:
I agree that for now change the interface for all providers is not the best if there is no more that support this. But what I meant really is in the line 14, that I think we should consider to reflect in the mocked message how adding the sender name would look like. I guess it would be something like this from the data used:
```
const mockNovuMessage = {
  to: 'test@test2.com',
  subject: 'test subject',
  html: '<div> Mail Content </div>',
  from: 'test <test@tet.com>',
  attachments: [
    { mime: 'text/plain', file: Buffer.from('dGVzdA=='), name: 'test.txt' },
  ],
};
```
Considering that you have tested it manually could you double check this would be the generated response when adding the send name?

Reply:
Actually the sendgrid provider test was failing:

- I have changed the object in `toHaveBeenCalledWith` because the from property will now return an object with email and name, I cannot change the mockNovuMessage because the function expects `IEmailOptions`.

- I added an other missing other property that was causing even the main branch to fail the sendgrid test 

```
customArgs: {
      id: undefined,
    }
```

This should have failed already when I created the PR but as I can see from the GH action the provider tests were skipped -> I think you guys need to have a look at it.

Reply:
As for the actual result, it looks like this now

![image](https://user-images.githubusercontent.com/6652818/193830819-8304be81-2e53-4ebc-91ef-f733a8b4f4ce.png)

Reply:
@p-fernandez

Reply:
> This should have failed already when I created the PR but as I can see from the GH action the provider tests were skipped -> I think you guys need to have a look at it.

Seems this action is not working as expected so we will try to address it here:
https://github.com/novuhq/novu/pull/1442
Thanks for reporting it.